code/code/misc/info.cc

```diff
@@ -5385,642 +5385,291 @@ void TBeing::describeTrapDamType(const TTrap *obj, int) const
-        strcat(buffer, "Non-targeted utility rituals:\n\r");
-        break;
-    }
+  auto getComponentString = [this](spellNumT spellNumber) {
+    auto comp = std::find_if(CompInfo.begin(), CompInfo.end(),
+                             [spellNumber](compInfo &ci) { return ci.spell_num == spellNumber; });
 
-    for (j = 0; j < skillSortVec.size(); j++) {
-      i = skillSortVec[j].theSkill;
-      das = getDisciplineNumber(i, FALSE);
-      if (das == DISC_NONE) {
-        vlogf(LOG_BUG, format("Bad disc for skill %d in doRituals") %  i);
-        continue;
-      }
-      cd = getDiscipline(das);
-      
-      // getLearnedness is -99 for an unlearned skill, make it seem like a 0
-      int tmp_var = ((!cd || cd->getLearnedness() <= 0) ? 0 : cd->getLearnedness());
-      tmp_var = min((int) MAX_DISC_LEARNEDNESS, tmp_var);
-
-      switch (type) {
-        case 0: // single target offensive
-          if (!(discArray[i]->targets & TAR_VIOLENT) ||
-              (discArray[i]->targets & TAR_AREA))
-            continue;
-          break;
-        case 1: // area offensive
-          if (!(discArray[i]->targets & TAR_VIOLENT) ||
-              !(discArray[i]->targets & TAR_AREA))
-            continue;
-          break;
-        case 2: // targeted utility
-          if ((discArray[i]->targets & TAR_VIOLENT) ||
-              !(discArray[i]->targets & TAR_CHAR_ROOM))
-            continue;
-          break;
-        case 3: // non-targeted utility
-          if ((discArray[i]->targets &  TAR_VIOLENT) ||
-              (discArray[i]->targets & TAR_CHAR_ROOM))
-            continue;
-          break;
-      }
+    if (comp != CompInfo.end() && comp->comp_num > 0) {
+      std::vector<TThing *> possibleComponentLocations = {
+          heldInPrimHand(),       heldInSecHand(),      this,
+          equipment[WEAR_WAIST],  equipment[WEAR_NECK], equipment[WEAR_WRIST_R],
+          equipment[WEAR_WRIST_L]};
 
-      // can't we say if !cd, continue here?
-      if (cd && !cd->ok_for_class && getSkillValue(i) <= 0) 
-        continue;
+      int totalCharges = 0;
+      for (TThing *location : possibleComponentLocations)
+        if (location)
+          totalCharges += findComponentCharges(location, spellNumber);
 
-      totalcharges = 0;
-      
-      for (l = 0; l < 7; l++) {
-        if (search[l].where && ritlevel >= search[l].ritlevel) {
-          totalcharges += findComponentCharges(search[l].where, i);
-        }
-      }
+      return format("   [%3i] %s") % totalCharges %
+             obj_index[real_object(comp->comp_num)].short_desc;
+    } else
+      return format("   No Component Found");
+  };
 
-      if ((getSkillValue(i) <= 0) &&
-          (!tmp_var || (discArray[i]->start - tmp_var) > 0)) {
-        if (!showall) 
+  sstring buffer = "";
+  auto bufferSpells = [this, &buffer, showUnlearned, getComponentString](std::vector<spellNumT> &skills,
+                                                     const sstring &spacing) {
```

Comment:
gah, what happened to the spaces here?

Reply:
Unrefined clang-format rules happened, I suspect. This code no longer exists in the newest version, however.